### ISSUE TITLE
NetworkError: 404 Not Found on _changes longpoll after new Hoodie instance

### ISSUE THREAD

**Issue by inator** [hoodiehq/hoodie#245]
See Issue #244 for reference.

App created using Version: 0.4.8 (node v0.10.24, npm 1.4.4, platform: linux)

Server runs error free, but when I try to log in a user from the frontend using:

```
    <body>
        <script src="http://code.jquery.com/jquery-1.10.1.min.js"></script>
        <script src="http://IP:PORT/_api/_files/hoodie.js"></script>
        <script>
            var app = new Hoodie('http://IP:PORT');

            //sign out
            app.account.signOut();

            //register a new user
            if (!app.account.username) app.account.signUp('joe@example.com', 'pass');

            //sign in the new user
            app.account.signIn('joe@example.com', 'pass')
            .done(function(data){
                console.log(data);
            })
            .fail(function(data){
                console.log(data);
            });
        </script>
    </body>
```

I am getting some 404's from a changes long poll that is established immediately following new Hoodie instance (please note... the referenced user hash does not match any known user).  A clear of local storage and cache yields no different result.  Nor does the successful sign in of a known user.

```
"NetworkError: 404 Not Found - http://ip:port/_api/user%2Fylivdux/_changes?include_docs=true&since=0&heartbeat=10000&feed=longpoll"
{"error":"not_found","reason":"no_db_file"}
```

The corresponding logged entry on the server is

```
Error getting update from changes pool
{ [Error: not_found
no_db_file]
  error: 'not_found',
  reason: 'no_db_file',
  response: 
...
```

**Comment by gr2m** [comment]
Your code doesn't make much sense to me. If you want a user to be signed in just do

``` js
var app = new Hoodie('http://IP:PORT');
app.account.signIn('joe@example.com', 'pass')
.done(function(data){
    console.log(data);
})
.fail(function(data){
    console.log(data);
});
```

**Comment by gr2m** [comment]
Also, beware that sessions (cookies, localStorage, ...) are shared across the same domain, despite different port numbers. That might cause trouble in your case as well

**Comment by inator** [comment]
@gr2m - Thanks very much for the reply and sorry for the confusing code.  I went with the simpler version and the 404's persist.

I'm sure I could be missing something elementary, but I never had this happen with the earlier versions of Hoodie (pre-January 2014).  Just for the sake of thoroughness, do you happen to have some standard front end code to run against a remote Hoodie instance for some basic tests?  If you do, would you mind sharing via Gist?  I figure I might as well use what you do if you have something like that.  Thanks much!

**Comment by gr2m** [comment]
I tried to develop some basic full stack tests to https://github.com/hoodiehq/hoodie-integration-test but I couldn't get it to work, see https://github.com/hoodiehq/hoodie-integration-test/issues/1#issuecomment-35112495

When you run it in a private browser tab, does the issue still persist? Can you reproduce it with the standard app coming from `hoodie new myappname`?

**Comment by inator** [comment]
@gr2m - I decided to start from scratch.  I deleted local storage & cache, deleted the user server side, and restarted the Hoodie instance for good measure.  I'm baffled by this, but for some reason it seems to be working even though I'm sure I tried this before - although I'm not positive I did all three together.  I was also working the CORS issue, so perhaps something carried over from that effort.

I'll close this and reopen if it comes up again.  Thanks for taking the time to pitch in!  Much appreciated!
